### PR TITLE
ID-1285 switch open api to only oauth2 security scheme

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: https://broadworkbench.atlassian.net/browse/ID-1285

  \<Don't forget to include the ticket number in the PR title!\>

What:

  \<For your reviewers' sake, please describe in a sentence or two what this PR is accomplishing (usually from the users' perspective, but not necessarily).\>

Why:

  \<For your reviewers' sake, please describe in ~1 paragraph what the value of this PR is to our users or to ourselves.\>

How:

  \<For your reviewers' sake, please describe in ~1 paragraph how this PR accomplishes its goal.\>

  \<If the PR is big, please indicate where a reviewer should start reading it (i.e. which file or function).\>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

File: docker/run-proxy.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ start() {
     docker create --name "${CONTAINER}" \
         --restart "always" \
         -p 40080:80 -p 50443:443 \
+        -e "B2C_APPLICATION_ID=${OIDC_CLIENT_ID}" \
```
Comment (dvoet) [inline docker/run-proxy.sh:14]: fixes local deployment to do the audience check correctly

File: project/Dependencies.scala [hunk 2]
```diff
@@ -82,22 +82,23 @@ object Dependencies {
 
   // All of workbench-libs pull in Akka; exclude it since we provide our own Akka dependency.
   // workbench-google pulls in workbench-{util, model, metrics}; exclude them so we can control the library versions individually.
+  val excludeBouncyCastle = ExclusionRule("org.bouncycastle")
```
Comment (dvoet) [inline project/Dependencies.scala]: bouncycastle got included in a few more places that need to be excluded

File: src/main/resources/swagger/api-docs.yaml [hunk 3]
```diff
@@ -4750,9 +4748,10 @@ components:
...
       x-tokenName: id_token
-    bearerAuth:
-      type: http
-      scheme: bearer
\ No newline at end of file
+      flows:
+        authorizationCode:
+          authorizationUrl: OAUTH_AUTHORIZATION_URL
+          scopes: {}
+          tokenUrl: OAUTH_TOKEN_URL
```
Comment (dvoet) [inline src/main/resources/swagger/api-docs.yaml:4757]: replacements for `OAUTH_AUTHORIZATION_URL` and `OAUTH_TOKEN_URL` were implemented in [workbenchLibs](https://github.com/broadinstitute/workbench-libs/pull/1684)

**Review by Ghost-in-a-Jar — APPROVED** [review]
thanks for doing this

**Review by tlangs — APPROVED** [review]

**Pull request merged**